books/tests/test_views.py

```diff
@@ -0,0 +1,131 @@
+        }
+
+        user_client
+        response = api_client.post(
+            self.endpoint,
+            body,
+            format='json',
+        )
+
+        assert response.status_code == status.HTTP_201_CREATED
+        assert "id" in json.loads(response.content)
+
+    def test_post_book_invalid(self, api_client,         user_client):
+        body_invalid = {
+            "title": "title1231222223",
+            "descriprion": "lorem lorem lorem lorem lorem lorem lorem lorem lorem lorem",
+            "phone_number": "asdasdasd"
+        }
+
+        user_client
+        response = api_client.post(
+            self.endpoint,
+            body_invalid,
+            format='json',
+        )
+
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+    def test_retrieve_book(self, book, api_client, user_client):
+
+        url = f'{self.endpoint}{book.id}/'
+
+        user_client
+        response = api_client.get(url)
+
+        assert isinstance(json.loads(response.content), dict)
+        assert Bookserializer(json.loads(response.content))
+        assert response.status_code == status.HTTP_200_OK
+
+    def test_update_book_success(self, book, api_client, user_client):
+        url = f'{self.endpoint}{book.id}/'
+        body_new_book = {
+            "id": book.id,
+            "title": "Updated",
+            "descriprion": "Updated",
+            "phone_number": book.phone_number,
+            "created_at": book.created_at.strftime('%Y-%m-%dT%H:%M:%S.%fZ'),
+            "updated_at": datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%fZ')
+        }
+
+        user_client
+        response = api_client.put(
+            url,
+            body_new_book,
+            format='json',
+        )
+
+        assert isinstance(json.loads(response.content), dict)
+        assert json.loads(response.content).get('title') == body_new_book.get('title')
+        assert response.status_code == status.HTTP_200_OK
+
+    def test_update_book_invalid(self, book, api_client, user_client):
+        url = f'{self.endpoint}{book.id}/'
+        body_new_book_invalid = {
+            "id": book.id,
+            "title": "Updated",
+            "descriprion": "Updated",
+            "phone_number": "asdfghj",
+            "created_at": book.created_at.strftime('%Y-%m-%dT%H:%M:%S.%fZ'),
+            "updated_at": datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%fZ')
+        }
+        user_client
+        response = api_client.put(
+            url,
+            body_new_book_invalid,
+            format='json',
+        )
+
+        assert isinstance(json.loads(response.content), dict)
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+    def test_delete_book(self, book, api_client, user_client):
+        url = f'{self.endpoint}{book.id}/'
+
+        user_client
+        response = api_client.delete(
+            url,
+            format='json',
+        )
+        assert response.status_code == 204
```

Comment:
Keep it consistent throughout the project. Should be `assert response.status_code == status.HTTP_204_NO_CONTENT`